instrumentation/github.com/go-chi/chi/splunkchi/option.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkchi
+
+import (
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/signalfx/splunk-otel-go/instrumentation/internal"
+)
+
+func localToInternal(opts []Option) []internal.Option {
+	out := make([]internal.Option, len(opts))
+	for i, o := range opts {
+		out[i] = internal.OptionFunc(func(c *internal.Config) { o.apply(c) })
+	}
+	return out
+}
+
+// Option applies options to a configuration.
+type Option interface {
+	apply(*internal.Config)
+}
+
+type optionConv struct {
+	iOpt internal.Option
+}
+
+func (o optionConv) apply(c *internal.Config) {
+	o.iOpt.Apply(c)
+}
+
+// WithTracerProvider returns an Option that sets the TracerProvider used for
+// a configuration.
+func WithTracerProvider(tp trace.TracerProvider) Option {
+	return optionConv{iOpt: internal.WithTracerProvider(tp)}
+}
+
+// WithAttributes returns an Option that appends attr to the attributes set
+// for every span created.
+func WithAttributes(attr []attribute.KeyValue) Option {
+	return optionConv{iOpt: internal.WithAttributes(attr)}
+}
+
+// WithPropagator returns an Option that sets p as the TextMapPropagator used
+// when propagating a span context.
+func WithPropagator(p propagation.TextMapPropagator) Option {
```

Comment:
Other SDKs generally set a global propagator. Do we prefer per instrumentatin/server propagator or is chi a special case?

Reply:
AFAIK it is just an option to override the global propagator

Reply:
> AFAIK it is just an option to override the global propagator

Correct, this will use the passed Propagator if this option is provided, otherwise it uses the globals.